packages/shared/src/components/search/SearchBar.spec.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * TESTS
+ * - should render with search history empty
+ * - should render with search history populated
+ * - should render with submit
+ * - should render with completion loading state
+ * - should render with the text suggestions
+ * - should render with the text for anon
+ * - should render with the text for first time login (no read history)
+ */
+
+import React from 'react';
+import { render, RenderResult, screen, waitFor } from '@testing-library/preact';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import nock from 'nock';
+import { AuthContextProvider } from '../../contexts/AuthContext';
+import loggedUser from '../../../__tests__/fixture/loggedUser';
+import { generateTestSquad } from '../../../__tests__/fixture/squads';
+import { SearchBar, SearchBarProps } from './SearchBar';
+
+beforeEach(async () => {
+  nock.cleanAll();
+  jest.clearAllMocks();
+});
+
+const squads = [generateTestSquad()];
+
+const renderComponent = (
+  loggedIn = true,
+  props: Partial<SearchBarProps> = {},
+): RenderResult => {
+  const client = new QueryClient();
+  const defaultProps: SearchBarProps = {
+    inputId: 'name',
+    name: 'name',
+  };
+
+  return render(
+    <QueryClientProvider client={client}>
+      <AuthContextProvider
+        user={loggedIn ? loggedUser : null}
+        updateUser={jest.fn()}
+        tokenRefreshed
+        getRedirectUri={jest.fn()}
+        loadingUser={false}
+        loadedUserFromCache
+        squads={squads}
+      >
+        <SearchBar {...defaultProps} {...props} />
+      </AuthContextProvider>
+    </QueryClientProvider>,
+  );
+};
+
+describe('SearchBar', () => {
+  it('should render with the beta flag', async () => {
+    renderComponent();
+
+    expect(screen.getByTestId('searchBar')).toBeInTheDocument();
+    expect(screen.getByPlaceholderText('Ask anything…')).toBeInTheDocument();
+    expect(screen.getByText('Beta')).toBeInTheDocument();
+  });
+
+  it('should render with no clear button when the value is empty', async () => {
+    renderComponent();
+    const clear = screen.queryByTitle('Clear query');
+
+    await waitFor(() => expect(clear).not.toBeInTheDocument());
+  });
+
+  it('should render with clear button when there is a value and clear on click', async () => {
+    renderComponent(true, { value: 'search' });
+    const input = screen.queryByRole('textbox') as HTMLInputElement;
+    const clear = screen.queryByTitle('Clear query');
+
+    await waitFor(() => expect(clear).toBeInTheDocument());
+    expect(input).toHaveValue('search');
+    clear.click();
```

Comment:
More of a general note, prefer `fireEvent` instead of `click` because it triggers events more like it happened in the browser. More [info](https://github.com/dailydotdev/apps/pull/2005/files). We could even use `userEvent` but those are better suited for e2e testing.